internal/cmd/controller/reconciler/bundledeployment_controller.go

```diff
@@ -139,7 +139,7 @@ func bundleDeploymentStatusChangedPredicate() predicate.Funcs {
 			if n == nil || o == nil {
 				return false
 			}
-			return !reflect.DeepEqual(n.Status, o.Status)
+			return !reflect.DeepEqual(n.Status, o.Status) || !n.DeletionTimestamp.IsZero()
```

Comment:
nit: swapping the conditions could prevent a call to `reflect.DeepEqual` if the bundle deployment has been marked for deletion (optimisation):
```suggestion
			return !n.DeletionTimestamp.IsZero() || !reflect.DeepEqual(n.Status, o.Status)
```